### DISCUSSION TITLE
PS3, Xbox 360, Wii U: Building, Booting, Supporting

### DISCUSSION THREAD

**Discussion by iamhumanipromise** [kth5/archpower#58] [General]
This post is a draft for now. Will format with more details later. Goal is to outline some information and make it into a mini alpha repo for knowledge about these three consoles.  

Right now Linux boots on all three platforms, and ArchPOWER boots on Wii U (from photo in other chat thread). 

Does any one know what the kernel version and support status is for: 

- **PS3**: version ??? via DVD-ROM, not mainlined | 1C/2T 512KB L2 + 8C/T SPE 256KB L2 | # of Cores Functional: ?? | Mesa Graphics Acceleration: OpenGL 2, Gallium 9 possible
- **Xbox 360**: Version 5.x, mainlined status = ?? | 3C/6T IBM Xenon 3.2GHz 32KB L1 1MB L2 / # Cores Functional: ?? | Mesa Graphics Acceleration: OpenGL 2, Gallium 9possible
- **Wii U:** Version 4.19; Out of Tree | IBM Espresso 3C/T 1.24GHz 32KB L1 512KB L2 / # of Cores Functional = 1x | Mesa Graphics Acceleration: OpenGL 4, Gallium 9, OpenCL, Vulkan Possible

I cannot seem to find anyone working on these: but I'm sure they are out there. Are those resources available for anyone interested? At least this page could act as waypoint to that knowledge, to hopefully get interested parties aimed the right way... then ArchPOWER on them and cruising. 

_**For GPU reverse engineering... referencing the history of the Asahi Linux folks may be worthwhile just for general technique. The GPUs are:**_ 

- **PS3:** Nvidia RSX "Reality Synthesizer" (NV47) / OpenGL ES 2.0 / 24 Pixel Shaders + 8 Vertex Shaders / 550MHz Core Clock / 256MB GDDR3 VRAM w/128MB Bus 
- **Xbox 360:** ATI "Xenos" (R500) / ~OpenGL 2.1/GLES 2.0/Shader Model 3.0 / 48 unified shaders + 16 texture units / 500MHz core clock / 10MB eDRAM + 512MB GDDR3 SoC shared access VRAM/RAM. 
- Wii U: AMD Radeon "Unknown" (RV770) / OpenGL ES 2.0; Shader Model 4.1 /OpenGL 4.0+/OpenGL ES 3.2/ 500MHz Core Clock / Shader Specs = ___?? / _?_ MB eDRAM RAM / 2GB shared DDR3 

The Wii U has the most potentially featureful GPU by being able to theoretically support OpenCL as well as Vulkan via the in-progress Terakan driver for R600/R700 GPUs. 

**Conclusion: If you are a dev looking to mainline Linux (and specifically Arch) for a console: consider the Wii U.** The gamepad support means you have a wireless display. 

_For now: anyone can get started with ArchPOWER and Linux 4.19 by __________ (placeholder for steps)_

**Comment by Fancy2209** [comment]
I have gotten PS3 booting with kernel 6.12
